// @kanelloc/react-native-animated-header-scroll-view/lib/commonjs/components/AnimatedHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AnimatedHeader = void 0;
var _reactNative = require("react-native");
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const AnimatedHeader = _ref => {
  let {
    HeaderComponent,
    headerImage,
    imageHeight,
    translateYUp,
    translateYDown,
    scale,
    imageStyle
  } = _ref;
  const {
    width
  } = (0, _reactNative.useWindowDimensions)();
  const AnimatedImageBackground = _reactNative.Animated.createAnimatedComponent(_reactNative.ImageBackground);
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.imgContainer, {
      marginTop: -imageHeight * 4,
      paddingTop: imageHeight * 4
    }]
  }, HeaderComponent ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, headerImage ? /*#__PURE__*/_react.default.createElement(AnimatedImageBackground, {
    source: headerImage,
    style: [{
      height: imageHeight,
      width: width * 1.2
    }, {
      transform: [{
        scale: scale
      }, {
        translateY: translateYUp
      }, {
        translateY: translateYDown
      }]
    }, imageStyle]
  }, HeaderComponent) : /*#__PURE__*/_react.default.createElement(_reactNative.Animated.View, {
    style: [{
      height: imageHeight,
      width: width * 1.2
    }, {
      transform: [{
        scale
      }, {
        translateY: translateYUp
      }, {
        translateY: translateYDown
      }]
    }]
  }, HeaderComponent)) : /*#__PURE__*/_react.default.createElement(_reactNative.Animated.Image, {
    source: headerImage,
    style: [{
      height: imageHeight,
      width: width * 1.2
    }, {
      transform: [{
        scale
      }, {
        translateY: translateYUp
      }, {
        translateY: translateYDown
      }]
    }, imageStyle]
  }));
};
exports.AnimatedHeader = AnimatedHeader;
const styles = _reactNative.StyleSheet.create({
  imgContainer: {
    alignItems: 'center',
    overflow: 'hidden'
  }
});
